### PR TITLE
Documented the new parseFile() method

### LINKED ISSUE #8424: Ability to parse YAML files directly
The YAML parser now offers a new `parseFile()` method to which you can pass the name of a YAML file instead of loadings its contents yourself as you would do when using the traditional `parse()` method (see symfony/symfony#24253).

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #8424.

### CODE REVIEW

File: components/yaml.rst [hunk 1]
```diff
@@ -102,13 +102,8 @@ string and converts it to a PHP array:
 
     use Symfony\Component\Yaml\Yaml;
 
-    $value = Yaml::parse(file_get_contents('/path/to/file.yml'));
-
-.. caution::
-
-    Because it is currently possible to pass a filename to this method, you
-    must validate the input first. Passing a filename is deprecated in
-    Symfony 2.2, and was removed in Symfony 3.0.
```
Comment (xabbuh) [inline components/yaml.rst]: we should already remove this block in the 3.3 docs

File: components/yaml.rst [hunk 2]
```diff
@@ -102,13 +102,8 @@ string and converts it to a PHP array:
...
+    $value = Yaml::parse("foo:\n    'bar'");
```
Comment (xabbuh) [inline components/yaml.rst]: having the newline here looks wrong to me, I would just use `$value = Yaml::parse('foo: bar');`

File: components/yaml.rst (hunk 1 above)
Comment (xabbuh) [reply components/yaml.rst]: see #8554

File: components/yaml.rst [hunk 3]
```diff
@@ -120,11 +115,28 @@ error occurred:
...
     } catch (ParseException $e) {
-        printf("Unable to parse the YAML string: %s", $e->getMessage());
+        printf('Unable to parse the YAML string: %s', $e->getMessage());
     }
 
+Reading YAML Files
+~~~~~~~~~~~~~~~~~~
+
+The :method:`Symfony\\Component\\Yaml\\Yaml::parseFile` method parses the YAML
+contents of the given file path and converts them to a PHP array:
```
Comment (xabbuh) [inline components/yaml.rst]: the result must not necessarily be an array

File: components/yaml.rst [hunk 4]
```diff
@@ -120,11 +115,28 @@ error occurred:
...
+
+.. code-block:: php
```
Comment (xabbuh) [inline components/yaml.rst]: can be omitted when terminating the previous paragraph with two colons

File: components/yaml.rst [hunk 5]
```diff
@@ -120,11 +115,28 @@ error occurred:
...
+
+    use Symfony\Component\Yaml\Yaml;
+
+    $value = Yaml::parseFile(file_get_contents('/path/to/file.yml'));
```
Comment (xabbuh) [inline components/yaml.rst]: `file_get_contents()` must be omitted here

**Comment by weaverryan** [comment]
Thank you @javiereguiluz.

File: components/yaml.rst [hunk 6]
```diff
@@ -120,11 +115,26 @@ error occurred:
...
 
+Reading YAML Files
+~~~~~~~~~~~~~~~~~~
+
+The :method:`Symfony\\Component\\Yaml\\Yaml::parseFile` method parses the YAML
+contents of the given file path and converts them to a PHP value::
+
+    use Symfony\Component\Yaml\Yaml;
+
+    $value = Yaml::parseFile('/path/to/file.yml');
```
Comment (apfelbox) [inline components/yaml.rst]: The method is not static https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Yaml/Parser.php#L49

Comment (xabbuh) [reply components/yaml.rst]: but it is static in the `Yaml` class (which is used here): https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Yaml/Yaml.php#L53 :)

Comment (apfelbox) [reply components/yaml.rst]: Ah, you're right!

**Pull request merged**